[DOCS] Removal of mentions of deprecated articles (#22509)
openvinotoolkit/openvino: docs/articles_en/documentation/openvino_legacy_features/mo_ovc_transition/legacy_conversion_api/supported_model_formats.rst

**MXNet, Caffe, and Kaldi** are legacy formats that need to be converted explicitly to OpenVINO IR or ONNX before running inference.

> OpenVINO **deprecated these formats** and **removed their support entirely**,

Reviewer: ```suggestion
As OpenVINO support for these formats has been discontinued,
```
The whole fragment should be changed to state clearly that these formats are not directly supported but can be used to obtain an ONNX model, or something like that

Author: Change applied in https://github.com/openvinotoolkit/openvino/pull/22509/commits/ef18601b07d035c40555cb80978cca7d3fb76669
